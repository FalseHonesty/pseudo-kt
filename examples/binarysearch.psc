function binarySearch(A, left, right, x)
    if left > right then
        return -1
    endif

    mid = (left + right) DIV 2
    if x == A[mid] then
        return mid
    elseif x < A[mid] then
        return binarySearch(A, left, mid - 1, x)
    else
        return binarySearch(A, mid + 1, right, x)
    endif
endfunction

A = {2, 5, 6, 8, 9, 10}
left = 0
right = A.length - 1

for i = 0 to 10
    index = binarySearch(A, left, right, i)
    if index != -1 then
        print("Element found at index " + str(index))
    else
        print("Element not found in the list")
    endif
next i
