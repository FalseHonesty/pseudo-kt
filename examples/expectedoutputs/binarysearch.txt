Element not found in the list
Element not found in the list
Element found at index 0
Element not found in the list
Element not found in the list
Element found at index 1
Element found at index 2
Element not found in the list
Element found at index 3
Element found at index 4
Element found at index 5
